homeassistant/components/emulated_hue/hue_api.py

```diff
@@ -179,6 +181,57 @@ def get(self, request, username):
         return self.json(json_response)
 
 
+class HueFullStateView(HomeAssistantView):
+    """Return full state view of emulated hue."""
+
+    url = "/api/{username}"
+    name = "emulated_hue:username:lights:state"
+    requires_auth = False
+
+    def __init__(self, config):
+        """Initialize the instance of the view."""
+        self.config = config
+
+    @core.callback
+    def get(self, request, username):
+        """Process a request to get the list of available lights."""
+        if not is_local(request[KEY_REAL_IP]):
+            return self.json_message("only local IPs allowed", HTTP_BAD_REQUEST)
+        if username not in HUE_API_USERNAME:
+            json_response = [
```

Comment:
Just return `self.json([…])` here so you can drop the `else:`

Reply:
Thank you for response, which is the best solution:
1. Have only one class for `HueAllLightsStateView` and `HueFullStateView`, and add the extra json data by example comparing from a the request.url (since URL for `HueAllLightsStateView` is `/api/{username}/lights` and URL for `HueFullStateView` is `/api/{username}`

```    
         if "lights" not in str(request.url):
            json_response = {
                "lights": json_response,
                "config": {
                    "mac": "00:00:00:00:00:00",
                    "swversion": "01003542",
                    "whitelist": {HUE_API_USERNAME: {"name": "HASS BRIDGE"}},
                    "ipaddress": str(self.config.advertise_ip)
                    + ":"
                    + str(self.config.advertise_port),
                },
            }
       return self.json(json_response)

```

or
2. Have the two different classes but creating a 
```
def get_all_lights_list(self.config, request):
        hass = request.app["hass"]
        json_response = {}

        for entity in hass.states.async_all():
            if config.is_entity_exposed(entity):
                state = get_entity_state(config, entity)
                number = config.entity_id_to_number(entity.entity_id)
                json_response[number] = entity_to_json(config, entity, state)
        return json_response
```